<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>515</width>
    <height>430</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QMaton</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Automaton settings</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="settingsWidget" native="true">
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="2">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>x</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QSpinBox" name="spWidth">
            <property name="suffix">
             <string> (width)</string>
            </property>
            <property name="maximum">
             <number>40</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Grid size:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spLength">
            <property name="suffix">
             <string> (length)</string>
            </property>
            <property name="maximum">
             <number>40</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Animation:</string>
            </property>
            <property name="buddy">
             <cstring>spinBox</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="3">
           <widget class="QSpinBox" name="spinBox">
            <property name="suffix">
             <string> images / second</string>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QToolButton" name="btnValidateSettings">
            <property name="toolTip">
             <string>Validate settings</string>
            </property>
            <property name="text">
             <string>Validate</string>
            </property>
            <property name="icon">
             <iconset resource="resources/resources.qrc">
              <normaloff>:/icons/check</normaloff>:/icons/check</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>16</width>
              <height>16</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QtVisualizer" name="wautomaton" native="true"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnPlay">
        <property name="toolTip">
         <string>Play / Pause automaton</string>
        </property>
        <property name="text">
         <string>Play / Pause</string>
        </property>
        <property name="icon">
         <iconset resource="resources/resources.qrc">
          <normaloff>:/media/play</normaloff>:/media/play</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnBack">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>One step backward</string>
        </property>
        <property name="text">
         <string>One step back</string>
        </property>
        <property name="icon">
         <iconset resource="resources/resources.qrc">
          <normaloff>:/media/backward</normaloff>:/media/backward</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnForward">
        <property name="toolTip">
         <string>One step forward</string>
        </property>
        <property name="text">
         <string>One step forward</string>
        </property>
        <property name="icon">
         <iconset resource="resources/resources.qrc">
          <normaloff>:/media/forward</normaloff>:/media/forward</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QSlider" name="slider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksBelow</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>515</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionReset"/>
    <addaction name="actionRandomizeGrid"/>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="fileToolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
  </widget>
  <widget class="QToolBar" name="editToolBar">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionReset"/>
   <addaction name="actionRandomizeGrid"/>
   <addaction name="actionClear"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/open</normaloff>:/actions/open</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="toolTip">
    <string>Open a JSON file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/save</normaloff>:/actions/save</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save to a JSON file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/close</normaloff>:/actions/close</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="toolTip">
    <string>Quit the application</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/reset</normaloff>:/actions/reset</iconset>
   </property>
   <property name="text">
    <string>R&amp;eset grid</string>
   </property>
   <property name="toolTip">
    <string>Reset grid as it was at step 1</string>
   </property>
  </action>
  <action name="actionRandomizeGrid">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/randomize</normaloff>:/actions/randomize</iconset>
   </property>
   <property name="text">
    <string>R&amp;andomize grid</string>
   </property>
   <property name="toolTip">
    <string>Reinitialize the grid with random statuses</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="resources/resources.qrc">
     <normaloff>:/actions/clear</normaloff>:/actions/clear</iconset>
   </property>
   <property name="text">
    <string>&amp;Clear</string>
   </property>
   <property name="toolTip">
    <string>Clear the automaton</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QtVisualizer</class>
   <extends>QWidget</extends>
   <header>visualizer/qt_visualizer.h</header>
   <container>1</container>
   <slots>
    <signal>started()</signal>
    <signal>finished()</signal>
    <signal>step_calculated()</signal>
    <slot>draw()</slot>
    <slot>run()</slot>
    <slot>stop()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>groupBox</tabstop>
  <tabstop>spLength</tabstop>
  <tabstop>spWidth</tabstop>
  <tabstop>spinBox</tabstop>
  <tabstop>btnValidateSettings</tabstop>
  <tabstop>btnPlay</tabstop>
  <tabstop>btnBack</tabstop>
  <tabstop>btnForward</tabstop>
  <tabstop>slider</tabstop>
 </tabstops>
 <resources>
  <include location="resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>_open_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>_save_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>_reset_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRandomizeGrid</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>_randomize_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>settingsWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>19</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>43</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wautomaton</sender>
   <signal>started()</signal>
   <receiver>MainWindow</receiver>
   <slot>_automaton_started()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>276</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wautomaton</sender>
   <signal>finished()</signal>
   <receiver>MainWindow</receiver>
   <slot>_automaton_finished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>514</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wautomaton</sender>
   <signal>step_calculated()</signal>
   <receiver>MainWindow</receiver>
   <slot>_automaton_step_calculated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>514</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnPlay</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>_start_pause_automaton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>39</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnForward</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>_run_forward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>190</x>
     <y>326</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>429</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>_clear_grid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_open_file()</slot>
  <slot>_save_file()</slot>
  <slot>_reset_grid()</slot>
  <slot>_randomize_grid()</slot>
  <slot>_automaton_started()</slot>
  <slot>_automaton_finished()</slot>
  <slot>_automaton_step_calculated()</slot>
  <slot>_start_pause_automaton()</slot>
  <slot>_run_forward()</slot>
  <slot>_clear_grid()</slot>
 </slots>
</ui>
